onadata/apps/api/permissions.py

```diff
@@ -62,6 +63,17 @@ class ExportDjangoObjectPermission(IsAuthenticated,
                                    ViewDjangoObjectPermissions):
 
     def has_permission(self, request, view):
+        export_pk = view.kwargs.get('pk', None)
```

Comment:
I think this check is in the wrong place, it should be at the `has_object_permission` level. `has_permission` only applies when you are requesting for the export list, in this case the request is for downloading a specific export.